src/listeners/message-listeners/index.ts

```diff
@@ -0,0 +1,37 @@
+import { Message, MessageResponse } from '../../messaging/scripts.messaging'
+import auth from './auth.listener'
+import locales from './locales.listener'
+import account from './account.listener'
+import test from './test.listener'
+
+const listenrs = [auth, locales, account, test]
+
+export function messageHandler(
```

Comment:
this handler is passed to the content listener that I guess answers for every event generated from any source.
here, the `auth` action surely should be called only from the extension's ui, so you should check the `sender` for that.

Reply:
yeah, well, I wouldn't say the login and register functions are dApp actions. it only should be called from the extension's popup page.